### PR TITLE
Add patch release

### PR DESCRIPTION
**Pull request by mortenjc**
Afonso is on vacation so here is a python release script that supports
major, minor and patch releases and supports updating itself.

Since I am not a python wizard maybe you'd want to have a look?

It follows (or should) the procedure described here
https://confluence.esss.lu.se/display/ECDC/Software+versioning+and+release+process+proposal

### CODE REVIEW

File: versioning/create-release.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+# Return tags as a string, replace newlines with space
+def git_get_tags():
+    return gitcmd('tag', 'failed to get tags').replace('\n', ' ')
+
+
+# used for both tags and VERSION file, strips leading 'v'
+# 'v1.2.3' -> [1, 2, 3] and '1.2.3' -> [1, 2, 3]
+def version_from_string(ver_str):
+    if len(ver_str) < 5:
```
Comment (SkyToGround) [inline versioning/create-release.py]: Why are we checking the length of the string here? What if there is a (manually created) version string that looks like this: `v1.2`?

File: versioning/create-release.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    print(res)
+
+# Return tags as a string, replace newlines with space
+def git_get_tags():
+    return gitcmd('tag', 'failed to get tags').replace('\n', ' ')
+
+
+# used for both tags and VERSION file, strips leading 'v'
+# 'v1.2.3' -> [1, 2, 3] and '1.2.3' -> [1, 2, 3]
+def version_from_string(ver_str):
```
Comment (SkyToGround) [inline versioning/create-release.py:72]: I am partial to solving problems like these with a regular expression. There are downsides to that solution as well of course.

File: versioning/create-release.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+from sh.contrib import git
+from sh import cat
+import os, sys, argparse, sh
+
+version_file="VERSION"
+
+# TODO implement
```
Comment (SkyToGround) [inline versioning/create-release.py]: What is it that needs to be done?

File: versioning/create-release.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        new = [ver[0], ver[1]+1, 0]
+    elif release == "patch":
+        new = [ver[0], ver[1], ver[2]+1]
+    else:
+        error_exit('Illegal relase (use major/minor/patch)')
+    return strings_from_version(new)
+
+#
+# command line options
+parser = argparse.ArgumentParser()
```
Comment (SkyToGround) [inline versioning/create-release.py]: I usually put the "main" execution code in a `main()` function. This code is then executed by placing the following two lines somewhere in the file:
```
if __name__ == '__main__':
    main()
```
I think this makes the code easier to read and easier to edit.

File: versioning/create-release.py (hunk 1 above)
Comment (mortenjc) [reply versioning/create-release.py]: just wanted to be sure that the string could at least contain 0.0.0
might switch to regex

File: versioning/create-release.py (hunk 2 above)
Comment (mortenjc) [reply versioning/create-release.py:72]: agree, considering switching to regex

File: versioning/create-release.py (hunk 3 above)
Comment (mortenjc) [reply versioning/create-release.py]: nothing, the comment will be deleted

File: versioning/create-release.py (hunk 4 above)
Comment (mortenjc) [reply versioning/create-release.py]: agree, will do

File: versioning/create-release.py (hunk 1 above)
Comment (mortenjc) [reply versioning/create-release.py]: switched to regex

File: versioning/create-release.py (hunk 4 above)
Comment (mortenjc) [reply versioning/create-release.py]: done

**Pull request merged**